File: app/src/main/java/com/privat/contacts/presentation/favorite/FavoriteFragment.java
package com.privat.contacts.presentation.favorite;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.privat.contacts.R;
import com.privat.contacts.databinding.FragmentFavoriteBinding;
import com.privat.contacts.presentation.base.BaseMvpView;
import com.privat.contacts.presentation.host.BottomNavigationHostFragmentDirections;
import com.privat.contacts.presentation.users.adapter.UsersAdapter;
import com.privat.contacts.presentation.users.model.UserItemUi;

import java.util.List;

import javax.inject.Inject;

public class FavoriteFragment extends BaseMvpView<FavoritePresenter> implements FavoriteView {
    @Inject
    FavoritePresenter favoritePresenter;
    private final UsersAdapter usersAdapter = new UsersAdapter(onClickId -> {
        Navigation.findNavController(requireActivity(), R.id.nav_host_fragment_activity_main)
                .navigate(
                        BottomNavigationHostFragmentDirections
                                .actionBottomNavigationHostFragmentToDetailsFragment(onClickId));
    }, favoriteClickId -> {
        favoritePresenter.changeFavorite(favoriteClickId);
    });
    private FragmentFavoriteBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentFavoriteBinding.inflate(inflater, container, false);
        binding.rvUsers.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.rvUsers.setAdapter(usersAdapter);
        ((AppCompatActivity) requireActivity()).getSupportActionBar().setTitle(R.string.favorite);
        DividerItemDecoration itemDecorator = new DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL);
        itemDecorator.setDrawable(ContextCompat.getDrawable(requireContext(), R.drawable.divider));
        binding.rvUsers.addItemDecoration(itemDecorator);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        favoritePresenter.clearTempData();
        favoritePresenter.startObserve();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
        favoritePresenter.stopObserve();
    }

    @Override
    protected FavoritePresenter presenter() {
        return favoritePresenter;
    }

    @Override
    public void showFavoriteList(List<UserItemUi> userItemUis) {
        if (binding != null)
            usersAdapter.updateData(userItemUis);
    }
}

File: app/src/main/java/com/privat/contacts/presentation/favorite/FavoriteFragment.kt
package com.privat.contacts.presentation.favorite

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.navigation.Navigation.findNavController
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.privat.contacts.R
import com.privat.contacts.databinding.FragmentFavoriteBinding
import com.privat.contacts.presentation.host.BottomNavigationHostFragmentDirections
import com.privat.contacts.presentation.users.UsersAdapter
import com.privat.contacts.presentation.users.model.UserItemUi
import dagger.android.support.DaggerFragment
import javax.inject.Inject

class FavoriteFragment : DaggerFragment(), FavoriteView {
    @Inject
    lateinit var favoritePresenter: FavoritePresenter
    private val usersAdapter = UsersAdapter({ onClickId: Int? ->
        findNavController(requireActivity(), R.id.nav_host_fragment_activity_main)
                .navigate(
                        BottomNavigationHostFragmentDirections
                                .actionBottomNavigationHostFragmentToDetailsFragment(onClickId!!))
    }) { favoriteClickId: Int -> favoritePresenter!!.changeFavorite(favoriteClickId) }
    private var binding: FragmentFavoriteBinding? = null
    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentFavoriteBinding.inflate(inflater, container, false)
        binding!!.rvUsers.layoutManager = LinearLayoutManager(requireContext())
        binding!!.rvUsers.adapter = usersAdapter
        (requireActivity() as AppCompatActivity).supportActionBar!!.setTitle(R.string.favorite)
        val itemDecorator = DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL)
        itemDecorator.setDrawable(ContextCompat.getDrawable(requireContext(), R.drawable.divider)!!)
        binding!!.rvUsers.addItemDecoration(itemDecorator)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        favoritePresenter!!.clearTempData()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        favoritePresenter!!.onAttach(this)
    }

    override fun onDetach() {
        super.onDetach()
        favoritePresenter!!.onDetach()
    }

    override fun showFavoriteList(userItemUis: List<UserItemUi>) {
        usersAdapter.updateData(userItemUis)
        usersAdapter.notifyDataSetChanged()
    }
}